### PR TITLE
fix #611 rename $repo and refactor channelcomparison

### LINKED ISSUE #611: Compare channels is not using Aurora's list of locales
Reported by Vanja for Serbian
https://transvision.mozfr.org/channelcomparison/

Serbian is missing because `$repo`, after being initialized to 'aurora', is re-set to 'central' in a following cycle.

### PR DESCRIPTION
**Pull request by SkySymbol**

### CODE REVIEW

File: app/models/channelcomparison.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+$strings = [];
+$strings[$chan1] = Utils::getRepoStrings($locale, $chan1);
+$strings[$chan2] = Utils::getRepoStrings($locale, $chan2);
+
+$chan_selector1 = $chan_selector2 = '';
+foreach ($desktop_repos as $repo_name) {
+    $ch1 = ($repo_name == $chan1) ? ' selected' : '';
+    $ch2 = ($repo_name == $chan2) ? ' selected' : '';
+    $chan_selector1 .= "\t<option" . $ch1 . " value=" . $repo . ">" . $repos_nice_names[$repo_name]. "</option>\n";
```
Comment (flodolo) [inline app/models/channelcomparison.php]: Fix the `$repo`, also following line

File: app/models/channelcomparison.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
+<?php
+namespace Transvision;
+
+require_once INC . 'l10n-init.php';
```
Comment (flodolo) [inline app/models/channelcomparison.php]: Already included from the controller

File: app/models/channelcomparison.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+// build the target locale switcher
+$target_locales_list = '';
+
+foreach ($loc_list as $loc) {
+    $ch = ($loc == $locale) ? ' selected' : '';
+    $target_locales_list .= "\t<option" . $ch . " value=" . $loc . ">" . $loc . "</option>\n";
+}
+
+$temp = array_intersect_key($strings[$chan1], $strings[$chan2]);
```
Comment (flodolo) [inline app/models/channelcomparison.php]: Let's use a better name for this variable, e.g. `$common_strings`.

**Comment by flodolo** [comment]
Nit: fix the typo in the commit message and the PR title (refactor)

File: app/views/channelcomparison.php [hunk 4]
```diff
@@ -1,46 +1,3 @@
...
-
-foreach ($loc_list as $loc) {
-    $ch = ($loc == $locale) ? ' selected' : '';
-    $target_locales_list .= "\t<option" . $ch . " value=" . $loc . ">" . $loc . "</option>\n";
-}
-
-$temp = array_intersect_key($strings[$chan1], $strings[$chan2]);
-$temp = array_diff($temp, $strings[$chan2]);
-?>
   <form id ="searchform" name="searchform" method="get" action="">
```
Comment (flodolo) [inline app/views/channelcomparison.php]: Fix the indentation of this (HTML).

Also note that the view is completely broken now ;-)

File: app/views/channelcomparison.php [hunk 5]
```diff
@@ -79,11 +40,12 @@
...
-         . "    <td><span class='celltitle'>Key</span><div class='string'>" . ShowResults::formatEntity($k) . "</div></td>\n"
-         . "    <td><span class='celltitle'>{$chan1}</span><div class='string'>" . ShowResults::highlight($v, $locale) . "</div></td>\n"
-         . "    <td><span class='celltitle'>{$chan2}</span><div class='string'>" . ShowResults::highlight($strings[$chan2][$k], $locale) . "</div></td>\n"
+         . "    <td><span class='celltitle'>Key</span><div class='string'>" . ShowResults::formatEntity($key) . "</div></td>\n"
+         . "    <td><span class='celltitle'>{$chan1}</span><div class='string'>" . ShowResults::highlight($value, $locale) . "</div></td>\n"
+         . "    <td><span class='celltitle'>{$chan2}</span><div class='string'>" . ShowResults::highlight($strings[$chan2][$key], $locale) . "</div></td>\n"
          . "  </tr>\n";
 }
 echo "</table>\n";
+?>
```
Comment (TheoChevalier) [inline app/views/channelcomparison.php]: I think we can remove this line

Comment (flodolo) [inline app/views/channelcomparison.php]: Yes, we actually should.

**Comment by flodolo** [comment]
It looks good now, thanks.

**Pull request merged**